src/App.js

```diff
@@ -45,6 +46,16 @@ function App() {
               }
               path={routes.PROFILE_PAGE}
             />
+            <Route
```

Comment:
instead of checking for token inside Route you can check for token outside and mention Route for success and failure.
for eg:
token 
? <Route success-route1 /> <Route success-route2/>
 : <Route failure-route1 /> <Route failure-route2/>

this looks cleaner and easier to take understand

Reply:
can you give an example show should I do it?